src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -589,5 +638,95 @@ private static string GetFullPath(K8SConfiguration configuration, string path)
                 return Path.Combine(Path.GetDirectoryName(configuration.FileName), path);
             }
         }
+
+        private static void MergeKubeConfig(K8SConfiguration basek8SConfig, K8SConfiguration mergek8SConfig)
+        {
+            // For scalar values, prefer local values 
+            basek8SConfig.ApiVersion = basek8SConfig.ApiVersion ?? mergek8SConfig.ApiVersion;
+            basek8SConfig.CurrentContext = basek8SConfig.CurrentContext ?? mergek8SConfig.CurrentContext;
+            basek8SConfig.FileName = basek8SConfig.FileName ?? mergek8SConfig.FileName;
+            basek8SConfig.Kind = basek8SConfig.Kind ?? mergek8SConfig.Kind;
+
+            // For Dictionaries, prefer first if not present in second.
+            if (mergek8SConfig.Preferences != null && mergek8SConfig.Preferences.Count > 0)
+            {
+                foreach (var preference in mergek8SConfig.Preferences)
+                {
+                    if (basek8SConfig.Preferences?.ContainsKey(preference.Key) == false)
+                    {
+                        basek8SConfig.Preferences[preference.Key] = preference.Value;
+                    }
+                }
+            }
+
+            if (mergek8SConfig.Extensions != null && mergek8SConfig.Extensions.Count > 0)
+            {
+                foreach (var extension in mergek8SConfig.Extensions)
+                {
+                    if (basek8SConfig.Extensions?.ContainsKey(extension.Key) == false)
+                    {
+                        basek8SConfig.Extensions[extension.Key] = extension.Value;
+                    }
+                }
+            }
+
+            // For IEnumerable, use name as unique identifier for whether we should insert or not.
+            if (mergek8SConfig.Contexts != null && mergek8SConfig.Contexts.Count() > 0)
+            {
+                var contexts = new List<Context>();
+                if (basek8SConfig.Contexts != null)
+                {
+                    contexts.AddRange(basek8SConfig.Contexts);
+                }
+
+                foreach (var context in mergek8SConfig.Contexts)
+                {
+                    if (basek8SConfig.Contexts?.FirstOrDefault(o => o.Name == context.Name) == null)
+                    {
+                        contexts.Add(context);
+                    }
```

Comment:
Probably want `else { throw ... }` so that people aren't confused if things are silently dropped.

Reply:
Hm, I'm curious about this. So if two kubeconfig.yml files define the same contexts (or Clusters, Users, Extensions or Preferences) or have them with the same name, it would be better to throw instead of ignore?

From what I can tell, that doesn't match what the go client does, but I'm not certain.

Reply:
I guess my point is that if people have multiple different contexts, I feel like they're going to be pretty confused no matter what. My recollection is that if there are two identical things the go client picks randomly because map iteration is random in Go, but I'm not positive.

Reply:
Based on that, I think picking the first one is fine then if the Go client picks randomly rather than throwing.